src/python/pants/testutil/pants_run_integration_test.py

```diff
@@ -557,42 +558,83 @@ def file_renamed(self, prefix, test_name, real_name):
             os.rename(real_path, test_path)
 
     @contextmanager
-    def temporary_file_content(self, path, content, binary_mode=True):
+    def temporary_directory_literal(
+        self, path: Union[str, Path],
+    ):
+        """Temporarily create the given literal directory, which must not exist."""
+        path = os.path.realpath(path)
+        assert path.startswith(
+            os.path.realpath(get_buildroot())
+        ), "cannot write paths outside of the buildroot!"
+        assert not os.path.exists(path), "refusing to overwrite an existing path!"
+
+        parent = os.path.dirname(path)
+        parent_ctx = (
+            suppress() if os.path.isdir(parent) else self.temporary_directory_literal(parent)
```

Comment:
I think you can use os.makedirs to get rid of the recursion and existence checks: https://docs.python.org/3/library/os.html#os.makedirs

Reply:
The recursion is to try to leave things as we found them: we recurse upward to create the parents temporarily, rather than permanently as with makedirs. Expanded the method docs.